atd-etl/socrata_export/socrata_export.py

```diff
@@ -1,121 +1,95 @@
-        "dataset_uid": ATD_ETL_CONFIG["SOCRATA_DATASET_PERSONS"],
+        "name": "people",
+        "typename": "socrata_export_people_view",
+        "dataset_id": SOCRATA_DATASET_PEOPLE,
+        "query": QUERIES["people"],
     },
 ]
 
-# Start timer
-start = time.time()
 
-# For each config, get records from Hasura and upsert to Socrata until res is []
-for config in query_configs:
-    print(f'Starting {config["table"]} table...')
-    is_truncate_complete = False
-    records = None
-    offset = 0
-    limit = 1000
-    total_records = 0
+def main(args_dict):
+    socrata_client = get_socrata_client()
 
-    # Query records from Hasura and upsert to Socrata
-    while records != []:
-        # Create query, increment offset, and query DB
-        query = config["template"].substitute(
-            limit=limit,
-            offset=offset,
-            date_limit=two_weeks_ago(),
-            initial_date_limit=ten_years_ago(),
-        )
-        offset += limit
-        data = run_hasura_query(query)
+    for dataset in datasets:
+        dataset_name = dataset["name"]
+        if not args_dict.get(dataset_name):
+            continue
 
-        if "data" not in data:
-            print("query: ", query)
-            print("data: ", data)
-            raise RuntimeError("There was a problem gathering data.")
+        logger.info(f"Processing {dataset_name}")
 
-        # Format records
-        records = config["formatter"](data, config["formatter_config"])
+        min_id = 0
+        records_processed = 0
+        while True:
+            logger.info(f"Fetching up to {RECORD_BATCH_SIZE} records")
+            variables = {"limit": RECORD_BATCH_SIZE, "minId": min_id}
+            records = make_hasura_request(query=dataset["query"], variables=variables)[
+                dataset["typename"]
+            ]
 
-        # truncate the existing data set once we have records
-        if records and not is_truncate_complete:
-            print(f'Truncating {config["table"]} table...')
-            client.replace(config["dataset_uid"], [])
-            is_truncate_complete = True
+            if min_id == 0:
+                if not records:
+                    raise Exception(
+                        "No records returned from Hasura. This should never happen"
+                    )
+                # we have some records, so it's safe to truncate the Socrata dataset
+                logger.info(f"Truncating dataset {dataset['dataset_id']}")
+                socrata_client.replace(dataset["dataset_id"], [])
+            elif not records:
+                logger.info(f"{records_processed} {dataset_name} records processed")
+                break
 
-        # Upsert records to Socrata
-        client.upsert(config["dataset_uid"], records)
-        total_records += len(records)
+            logger.info(
+                f"Upserting {len(records)} records IDs: {records[0]['id']} - {records[-1]['id']} (total records: {records_processed})"
```

Comment:
are we upserting tho or just completely replacing?

Reply:
jk i see we are upserting later i just got confused by how we replace at first?

Reply:
When min_id is 0 it replaces the whole dataset with an empty `[]` , and then does an upsert on every run until we run out of records, which is why the break happens on line 63. 

Could the logger say "(records already processed: {records_processed}" instead of "total records"? since its not the total number of records. 
`Upserting 1000 records IDS: 1-1000 (total records: 0)` is what I anticipate will log on the first run, right?

Reply:
Thanks for suggesting the logging improvements. And I will add some comments to the code that make the logic for querying records and replacing the dataset more clear 🙏